### ISSUE TITLE
WARNING: Failed to find app matching setting "appService.defaultWebAppToDeploy" with id "/subscriptions/..."

### ISSUE THREAD

**Issue by tonakai** [microsoft/vscode-azureappservice#2274]
Does this occur consistently?  Yes
Repro steps:

1. setup defaultwebapptodeploy in settings.json in .vscode folder: `"appService.defaultWebAppToDeploy": "/subscriptions/....",` this id is matching the id in the properties of this resource which is a app service slot.
2. i also try to set the prod as default but it is still unable to match the app.
3. right click on the empty space in explorer then select 'deploy to web app...'
4. instead of asking directly deploy to app service slot, it is asking me to select a web app. this used to be working.

let me know if you need some more information.

Version: 0.24.2
OS: win32
OS Release: 10.0.22000
Product: Visual Studio Code
Product Version: 1.68.1
Language: en

**Comment by nturinski** [comment]
Thanks for reporting this. We recently changed our tree view and I think that broke this line: https://github.com/microsoft/vscode-azuretools/blob/main/appservice/src/deploy/getDeployNode.ts#L38

**Comment by nturinski** [comment]
@tonakai Rather than setting up the `defaultwebapptodeploy` setting on your own, you should be able to right-click deploy to the web app. If the setting is cleared, a prompt should appear on the bottom right. Click "Yes", it should save the proper ID needed for the setting to work.

Just a workaround in the meantime!

![image](https://user-images.githubusercontent.com/5290572/176794905-1a575f27-913c-4bff-b84a-68630c6f9217.png)

**Comment by tonakai** [comment]
hi there @nturinski thanks for the help, initially I didn't set it up manually, I did the same steps as you mentioned (only I did it for the slots) but since the path changed it didn't want to update I guess. 

the new id is a bit verbose isn't it? it looks like this: 

`/subscriptions/<subscription-id>/microsoft.web/sites/subscriptions/<subscription-id>/resourceGroups/<resource-group-name>/providers/Microsoft.Web/sites/<app-name>/subscriptions/<subscription-id>/resourceGroups/<resource-group-name>/providers/Microsoft.Web/sites/<app-name>/slots/subscriptions/<subscription-id>/resourceGroups/<resource-group-name>/providers/Microsoft.Web/sites/<app-name>/slots/<slot-name>`

**Comment by nturinski** [comment]
@tonakai Yeah, agreed it is quite long.  This is a byproduct of some changes to our UI

We plan to change this behavior so that the `defaultWebAppToDeploy` id is the Azure Resource id (what it was originally), but that change may take some time as we need to fix some behaviors to our UI.
